src/crimson/os/seastore/cache.h

```diff
@@ -1020,7 +1019,7 @@ class Cache {
   ) {
     assert(extent->state == CachedExtent::extent_state_t::CLEAN_PENDING);
     extent->set_io_wait();
-    return reader.read(
+    return epm.read(
```

Comment:
This seems to be right to read multiple devices in Cache. Other changes looks good to me. 

Although I'm fine as is, one thing I'm not sure is the abstraction class by the name of "Device".
EPM has "Device" classes to read contents from segment manager or RBM. But, it only provides a read operation even if "Device" is a general term and it contains mkfs and mount.

Reply:
> But, it only provides a read operation even if "Device" is a general term and it contains mkfs and mount.

Do you mean that we can further refactor EPM to also include mkfs and mount interfaces?

BTW, `Device` is not only referenced by EPM, but also referenced by TransactionManager and SeaStore classes in the general paths. `Device` is assumed to be the parent class for both SegmentManager and RandomBlockManager. See discussions https://github.com/ceph/ceph/pull/45574#discussion_r843431596

Reply:
Hm.. Understood.